### PR TITLE
[Qt] show conflicts of unconfirmed transactions in the UI

### PR DESCRIPTION
**Pull request by jonasschnelli**
Currently, the GUI shows only conflicts with transaction in a block.

This results in displaying replaced transaction (RBF) wrong (the replaced transaction is no longer in the mempool, but still in the wallet-transactionlist, conflicts are not detected).

This PR will mark unconfirmed transactions that are **not** in our mempool and have conflicts as conflicted.

### CODE REVIEW

**Comment by jonasschnelli** [comment]
Another option would be to hide these transactions completely.

**Comment by jonasschnelli** [comment]
Before this PR:

Sending 49.9BTC and RBF with a 40.0BTC transaction.
bildschirmfoto 2016-04-06 um 17 17 40

After this PR:
bildschirmfoto 2016-04-06 um 17 21 22

File: src/qt/transactionrecord.cpp [hunk 1]
```diff
@@ -257,7 +261,21 @@ void TransactionRecord::updateStatus(const CWalletTx &wtx)
...
+        // transaction if the mempool status has been changed.
+        // Could be further improved by a signal fired when a
+        // mempool transaction has been evicted
+
+        LOCK(mempool.cs);
+        if (status.inMempool != mempool.exists(hash))
+            return true;
+    }
+
+    if (status.cur_num_blocks != chainActive.Height())
```
Comment (MarcoFalke) [inline src/qt/transactionrecord.cpp]: Nit: I think you can leave those three lines the way it was before: `return status.cur_num_blocks != chainActive.Height();`

**Comment by MarcoFalke** [comment]
>  Another option would be to hide these transactions completely.

I don't think you want to hide wtx without telling the user/ without having the user told you to do so.

File: src/qt/transactionrecord.cpp (hunk 1 above)
Comment (jonasschnelli) [inline src/qt/transactionrecord.cpp]: Right! Fixed.

**Comment by jonasschnelli** [comment]
> I don't think you want to hide wtx without telling the user/ without having the user told you to do so.

Yes. I agree. IMO the "mark as conflicted" solution is better.

**Comment by MarcoFalke** [comment]
Hmm, so I have a non-rbf transaction back from august:

```
Status: 0/unconfirmed, not in memory pool
Date: 8/19/15 15:51
```

and sendrawt returns `Missing inputs (code -25)`, but it is not showing the new icon. Need to look into this...

**Comment by jonasschnelli** [comment]
@MarcoFalke: does you non-rbf transaction back from august is in conflict with another transaction in your wallet? Only if it's so, the conflicted state icon will be shown.

**Comment by MarcoFalke** [comment]
No, it didn't conflict within the wallet.

Tested ACK 614073f

**Comment by MarcoFalke** [comment]
One could argue to use another status/icon for this e.g. https://raw.githubusercontent.com/jonasschnelli/bitcoin/47196712ec7f3ff2f5f814650d40366cb269fe56/src/qt/res/icons/transaction_replaceable.png with the conflicted icon (cross) in the lower left.

**Comment by jonasschnelli** [comment]
Rebased and changed the icon for transactions with conflict but not being in the mempool (RBF)l:
bildschirmfoto 2016-05-20 um 17 48 22

**Comment by luke-jr** [comment]
These seem to be two list items referring to the same transaction.

**Comment by jonasschnelli** [comment]
@luke-jr: this PR does not focus on hiding out transactions from the wallet (a lot more complexity).
At the moment, a replaced RBF transaction is not listed as conflicted (https://github.com/bitcoin/bitcoin/pull/7826#issuecomment-206424972).

This PR is a tiny fix to improve the GUI RBF attribution.

**Comment by jonasschnelli** [comment]
Another solution would be (idea once shortly discusses with @gmaxwell on IRC):
* directly hide transactions if they are conflicted and **not** in the mempool (RBF)
* hide transactions that conflicts with a transaction in a block >6
* add option in the settings panel to "display old conflicted transaction" and turn it off by default.

**Comment by sipa** [comment]
@jonasschnelli Any progress here?

**Comment by jonasschnelli** [comment]
@sipa: I'm working on a option to auto-hide mempool conflicts (transactions that are not in the blockchain and mempool) (RBF/abandoned txes).

**Comment by jonasschnelli** [comment]
I think this solution is fine as it is. The auto-hide option can be written in a separate PR.
This PR would be a simple change to allow users to easy identify mempool replacments relevant to the wallet.

Would be great for 0.13.

**Comment by laanwj** [comment]
Removing this from the 0.13 milestone as it missed the feature freeze.

> Would be great for 0.13.

Agree, would have been nice, just like #7817. I think we have a lack of GUI testers/reviewers, as useful GUI changes like this linger way too long.

**Comment by jonasschnelli** [comment]
This (simple) PR looks for reviewers...

**Comment by MarcoFalke** [comment]
I think the problem is that it may not be obvious for everyone  to set up  the appropriate transactions to test this. I think it may help to add "proposed steps to reproduce" for harder to test pulls in general.

**Comment by MarcoFalke** [comment]
Works for standard transactions:

screenshot from 2016-08-10 10-37-01
screenshot from 2016-08-10 10-37-06
screenshot from 2016-08-10 10-37-14

Does not work for non-standard transactions:

screenshot from 2016-08-10 10-34-42
screenshot from 2016-08-10 10-34-45

**Comment by jonasschnelli** [comment]
Closing for now due to inactivity...

**Pull request closed without merging**